### PR TITLE
Set content of CL_DEVICE_NAME and CL_DEVICE_VENDOR to meaningful valu…

### PR DESCRIPTION
**Pull request by blu**
…es for __arm__ and __aarch64__ host targets

### CODE REVIEW

File: lib/CL/devices/cpuinfo.c [hunk 1]
```diff
@@ -295,6 +332,19 @@ pocl_cpuinfo_get_cpu_name_and_vendor(cl_device_id device)
...
       break;
+
+#if __arm__ || __aarch64__
+    unsigned vendor_id;
+    if (1 == sscanf(start, "%x", &vendor_id)) {
+      for (size_t i = 0; i < sizeof(vendor_list) / sizeof(vendor_list[0]); ++i) {
+        if (vendor_id == vendor_list[i].id) {
+          start = vendor_list[i].name;
+          end = start + strlen (vendor_list[i].name);
+        }
```
Comment (pjaaskel) [inline lib/CL/devices/cpuinfo.c]: Should there be a 'break' here?

File: lib/CL/devices/cpuinfo.c [hunk 2]
```diff
@@ -313,12 +363,23 @@ pocl_cpuinfo_get_cpu_name_and_vendor(cl_device_id device)
...
     return;
 
+#if __arm__ || __aarch64__
+  unsigned part_id;
+  if (1 == sscanf(start, "%x", &part_id)) {
+    for (size_t i = 0; i < sizeof(part_list) / sizeof(part_list[0]); ++i) {
+      if (part_id == part_list[i].id) {
+        start = part_list[i].name;
+        end = start + strlen (part_list[i].name);
+      }
```
Comment (pjaaskel) [inline lib/CL/devices/cpuinfo.c]: Same.

File: lib/CL/devices/cpuinfo.c (hunk 1 above)
Comment (blu) [reply lib/CL/devices/cpuinfo.c]: Oops. Definitely.

**Comment by blu** [comment]
@pjaaskel, updated as per your review, and also coupled vendors and part lists (so far just ARM and APM).

**Pull request merged**